doc/development/guide/pullrequests.rst

```diff
@@ -42,16 +42,30 @@ merged.
 
       black -l 100 pennylane tests
 
+* **Python imports check**: We use `isort <https://pycqa.github.io/isort/>`_ to automatically sort Python imports.
+  You may run ``isort`` locally with ``make format``. ``isort`` will run with the following options: 
+
+  * ``--py 311`` tells to ``isort`` to reorder imports as if your interpreter were Python 3.11;
```

Comment:
Thoughts on using a similar CLI argument for `black`? I found [this](https://black.readthedocs.io/en/stable/usage_and_configuration/the_basics.html#t-target-version) argument which we can use to specify python versions. I notice sometimes that black formatting done locally causes the format check in CI to fail.

Reply:
In my experience, the Python version does not change `black` results, maybe because it is much more opinionated than `isort`. If you have observed changes between Python versions however, pinning the Python version would likely help. In any case, for `isort` it is necessary. Note that `--py 311` does not require Python 3.11, it is simply a lint option and can run with any Python interpreter.

Reply:
> I notice sometimes that black formatting done locally causes the format check in CI to fail.

I think another case when this happens is when the pre-commit hook version is not the same as the local or remote version.

Reply:
So would you prefer we also specify supported Python versions in `black` options in this PR?

Reply:
I think it's probably best to leave this for now. Don't want to block this PR.